src/test/java/com/synopsys/integration/detect/battery/BatteryTest.java

```diff
@@ -0,0 +1,319 @@
+        }
+
+        for (final String resourceFileName : resourceFileNames) {
+            final InputStream inputStream = BatteryFiles.asInputStream("/" + resourcePrefix + "/" + resourceFileName);
+            final File file = new File(sourceDirectory, resourceFileName);
+            Assertions.assertNotNull(inputStream, "Could not find resource file: " + file);
+            FileUtils.copyInputStreamToFile(inputStream, file);
+        }
+
+        for (final String resourceZipFileName : resourceZipNames) {
+            final File zipFile = BatteryFiles.asFile("/" + resourcePrefix + "/" + resourceZipFileName + ".zip");
+            final File target = new File(sourceDirectory, resourceZipFileName);
+            ZipUtil.unpack(zipFile, target);
+        }
+    }
+
+    private void assertBdio() throws IOException, JSONException {
+        final File[] bdio = bdioDirectory.listFiles();
+        Assertions.assertTrue(bdio != null && bdio.length > 0, "Bdio output files could not be found.");
+
+        if (shouldExpectBdioResources) {
+            final File expectedBdioFolder = BatteryFiles.asFile("/" + resourcePrefix + "/bdio");
+            final File[] expectedBdioFiles = expectedBdioFolder.listFiles();
+            Assertions.assertTrue(expectedBdioFiles != null && expectedBdioFiles.length > 0, "Expected bdio resource files could not be found: " + expectedBdioFolder.getCanonicalPath());
+            Assertions.assertEquals(expectedBdioFiles.length, bdio.length, "Detect did not create the expected number of bdio files.");
+
+            final List<File> actualByName = Arrays.stream(bdio).sorted(Comparator.comparing(File::getName)).collect(Collectors.toList());
+            final List<File> expectedByName = Arrays.stream(expectedBdioFiles).sorted(Comparator.comparing(File::getName)).collect(Collectors.toList());
+
+            for (int i = 0; i < expectedByName.size(); i++) {
+                final File expected = expectedByName.get(i);
+                final File actual = actualByName.get(i);
+                Assertions.assertEquals(expected.getName(), actual.getName(), "Bdio file names did not match when sorted.");
+
+                final String expectedJson = FileUtils.readFileToString(expected, Charset.defaultCharset());
+                final String actualJson = FileUtils.readFileToString(actual, Charset.defaultCharset());
+
+                final JSONArray expectedJsonArray = (JSONArray) JSONParser.parseJSON(expectedJson);
+                expectedJsonArray.getJSONObject(0).remove("creationInfo");
+                expectedJsonArray.getJSONObject(0).remove("@id");
+
+                final JSONArray actualJsonArray = (JSONArray) JSONParser.parseJSON(actualJson);
+                actualJsonArray.getJSONObject(0).remove("creationInfo");
+                actualJsonArray.getJSONObject(0).remove("@id");
+
+                JSONAssert.assertEquals(expectedJsonArray, actualJsonArray, false);
+            }
+
+        } else {
```

Comment:
Seems brittle. Not sure what else you can do here. I am thinking of the upcoming BDIO 2 upgrade.

Reply:
I don't want to invest too much time comparing BDIO because of BDIO2, this is a pretty 'low effort' BDIO compare because of that. When BDIO2 comes out I plan to add a gradle task to automatically update all the bdio.